File: cracking_the_coding_interview/1.1isUniqueChars.cs
using System;

class Solution
{
    static void Main(string[] args)
    {       
        Console.WriteLine(isUniqueChars("abc"));
    }
    
    static bool isUniqueChars(string str) {
        if (str.Length > 256) return false;
        bool[] check = new bool[256];
        foreach (char ch in str)
        {
            int idx = (int)ch;
            if (check[idx]) return false;
            else check[idx] = true;
        }
        return true;
    }
}

File: cracking_the_coding_interview/1.1isUniqueChars.vb
Module Solution
    Sub Main()
        Console.WriteLine (isUniqueChars("aabc"))
    End Sub

    Function isUniqueChars(str As String) as Boolean
        Dim strCheck(256) As Boolean = new Boolean(){}

        For Each ch As Char In str
            If strCheck(Asc(ch))=True Then
                isUniqueChars = False
                Exit Function
            Else
                strCheck(Asc(ch)) = True
            End If
        Next
        
        isUniqueChars = True
    End Function

End Module